### DISCUSSION TITLE
HostRules are not being set

### DISCUSSION THREAD

**Discussion by athreya92** [renovatebot/renovate#14493] [Request Help] [answered]
Hi,

A huge shout-out to you guys for building a great software. I'm trying to understand how renovatebot works. 

I'm trying to run renovatebot as docker to create onboarding PR. The PR is created successfully, but I see there are some dependency lookup failures messages on the PR. The repo has private go deps. So I figured, hostRules to be set in the config.js.

Even after adding it, the dependency failures did not disappear. 

I even did a dryRun with ```printConfig: true``` option to see the config being used. All i can see is hostRules are empty - ```hostRules: []```
Cant figure out whats wrong. 

renovate version - ```31.97.1```
config.js
```
module.exports = {
  endpoint: 'https://gitscm-server-xxx',
  platform: 'bitbucket-server',
  username: 'xxx',
  password: 'xxx',
  logLevel: 'debug',
  onboardingConfig: {
    extends: ['config:base'],
  },
  repositories: ['xxx/xxx'],
  hostRules: [
    {matchHost: 'gitscm-server-xxx', hostType: 'bitbucket-server', token:'xxx'} // token same as password?
  ],
  constraints: {
    go: '1.17'
  },
  postUpdateOptions: ['gomodTidy1.17'],
  dryRun: true,
  printConfig: true,
  requireConfig: true,
};
```

**Comment by rarkins** [comment, marked as answer]
Private Go Modules are not supported on all hosts yet, unfortunately. There's a couple of issues related to private modules open already if you browse https://github.com/renovatebot/renovate/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Amanager%3Agomod but if none cover your use case then feel free to raise a new one

**Reply by athreya92** [reply]
If I understand correctly, private go modules are not supported on ```bitbucket-server``` platform? so the hostRules are empty?

**Reply by rarkins** [reply]
I think there's two parts:
1. Supporting bitbucket-server as a host when we look up Go Modules
2. Passing credentials to Go when we run `go x`

Possibly both of the above are lacking.

**Reply by athreya92** [reply]
Thanks @rarkins for the info. Is there any roadmap planned for this support that I can watch out for?

**Reply by rarkins** [reply]
You can browse through the list of issues linked above and subscribe to any relevant. If none capture your problem and you have time to create a reproduction repository, then you are also welcome to create new bug report issues too.

**Reply by Shegox** [reply]
The empty hostRule happen because as part of the startup process Renovate analysis them, adds them to the global hostRules object and removes them from the config. 
https://github.com/renovatebot/renovate/blob/b0d4012c453e17c24ec3b041fa0f31173e92f2ce/lib/workers/global/index.ts#L144-L149
When the config is later printed, they are no longer represented.

**Reply by viceice** [reply]
this is expected behavior
